### ISSUE TITLE
"Auto Scrolling" TextArea removes first line instead of just growing

### ISSUE THREAD

**Issue by LennartH** [libgdx/libgdx#4561]
#### Issue details
I tried to build an auto scrolling TextArea (for an in-game log) based on [this snippet](https://github.com/libgdx/libgdx/issues/3796#issuecomment-202729107). But when text is added consecutively, the TextArea removes the first line instead of just growing, so that the outer ScrollPane can do its magic.

A simple workaround, was to just add 1 to the value returned by `getLines()`, but I don't like magic numbers and I think there's something wrong in the way the backing `lineBreaks` array is maintained.
I tried calling `calculateOffsets()` before the call of `getLines()`, but that didn't help.

#### SSCCE
```java
import com.badlogic.gdx.ApplicationAdapter;
import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.backends.lwjgl.LwjglApplication;
import com.badlogic.gdx.backends.lwjgl.LwjglApplicationConfiguration;
import com.badlogic.gdx.graphics.GL20;
import com.badlogic.gdx.scenes.scene2d.Actor;
import com.badlogic.gdx.scenes.scene2d.Stage;
import com.badlogic.gdx.scenes.scene2d.ui.Button;
import com.badlogic.gdx.scenes.scene2d.ui.ScrollPane;
import com.badlogic.gdx.scenes.scene2d.ui.Skin;
import com.badlogic.gdx.scenes.scene2d.ui.Table;
import com.badlogic.gdx.scenes.scene2d.ui.TextArea;
import com.badlogic.gdx.scenes.scene2d.ui.TextButton;
import com.badlogic.gdx.scenes.scene2d.utils.ChangeListener;
import com.badlogic.gdx.utils.viewport.ScreenViewport;

public class AutoScrollingTextAreaPrototype extends ApplicationAdapter {
    Stage stage;
    int lineCounter;

    public void create () {
        stage = new Stage(new ScreenViewport());
        Skin skin = new Skin(Gdx.files.internal("skin/default/uiskin.json"));
        Gdx.input.setInputProcessor(stage);

        Table container = new Table();
        stage.addActor(container);
        container.setFillParent(true);
        container.pad(10).defaults().expandX().fillX().space(4);

        final OpenScrollPane scrollPane = new OpenScrollPane(null, skin);
        scrollPane.setFadeScrollBars(false);
        scrollPane.setFlickScroll(false);
        scrollPane.setScrollingDisabled(true, false);

        final TextArea textArea = new TextArea("Initial line.\n"
                // Adding long text for soft line breaks
                + "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. "
                + "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. "
                + "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. "
                + "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. "
                + "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.", skin) {
            public float getPrefHeight () {
                float prefHeight = getLines() * getStyle().font.getLineHeight();
//                float prefHeight = (getLines() + 1) * getStyle().font.getLineHeight(); // Work around
                TextFieldStyle style = getStyle();
                if (style.background != null) {
                    prefHeight = Math.max(prefHeight + style.background.getBottomHeight() + style.background.getTopHeight(), style.background.getMinHeight());
                }
                return prefHeight;
            }
        };
        textArea.setDisabled(true);
        
        scrollPane.setWidget(textArea);

        Button addLineButton = new TextButton("Add new line", skin);
        addLineButton.addListener(new ChangeListener() {
            @Override
            public void changed(ChangeEvent event, Actor actor) {
                textArea.appendText("\nLine " + lineCounter++);
//                textArea.setCursorPosition(0); // Another work around
                scrollPane.invalidate();
                scrollPane.scheduleScrollToBottom(); // See OpenScrollPane below
            }
        });
        container.add(addLineButton).colspan(2);
        
        container.row().height(350);
        container.add(scrollPane);
        container.debugAll();
    }

    public void render () {
        Gdx.gl.glClear(GL20.GL_COLOR_BUFFER_BIT);
        stage.act(Gdx.graphics.getDeltaTime());
        stage.draw();
    }

    public void resize (int width, int height) {
        // Pass false to not modify the camera position.
        stage.getViewport().update(width, height, true);
    }

    public static void main (String[] args) throws Exception {
        LwjglApplicationConfiguration config = new LwjglApplicationConfiguration();
        config.width = 1024;
        config.height = 768;
        new LwjglApplication(new AutoScrollingTextAreaPrototype(), config);
    }
    
    public class OpenScrollPane extends ScrollPane {
        
        private boolean scrollToBottom;

        public OpenScrollPane(Actor widget, Skin skin) {
            super(widget, skin);
        }
        
        public void scheduleScrollToBottom() {
            scrollToBottom = true;
        }

        @Override
        public void layout() {
            super.layout();
            if (scrollToBottom) {
                setScrollY(getMaxY());
            }
        }
        
    }
}
```

#### Version of LibGDX and/or relevant dependencies
1.9.6-SNAPSHOT

#### Affected platforms
I assume all, but it was only tested on Windows.

**Comment by LennartH** [comment]
While working on my project, I found another work around for this problem, by setting the text areas curser position to 0, after adding the text. I adjusted the minimal example.
This is also a workaround for #4570.

**Comment by mgsx-dev** [comment]
@LennartH as per setScrollY (see below), it's not the right method to call. You should call scrollTo instead. In your case you can call setScroll(0,0,0,0) in order to scroll down. For other cases you have to validate scrollPane first in order to have a fresh valid clientArea size.

> Called whenever the y scroll amount is changed

**Comment by NathanSweet** [comment]
Should this issue be closed by #5520?
